src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/PackageRestoreConfiguredInputDataSource.cs

```diff
@@ -61,11 +61,12 @@ protected override IDisposable LinkExternalInput(ITargetBlock<RestoreUpdate> tar
             return transformBlock;
         }
 
-        private static PackageRestoreConfiguredInput CreateRestoreInput(ProjectConfiguration projectConfiguration, IImmutableDictionary<string, IProjectRuleSnapshot> update)
+        private static PackageRestoreConfiguredInput CreateRestoreInput(IProjectVersionedValue<IProjectSubscriptionUpdate> projectSubscriptionUpdate, ProjectConfiguration projectConfiguration, IImmutableDictionary<string, IProjectRuleSnapshot> update)
         {
             var restoreInfo = RestoreBuilder.ToProjectRestoreInfo(update);
+            projectSubscriptionUpdate.DataSourceVersions.TryGetValue(ProjectDataSources.ConfiguredProjectVersion, out IComparable configuredProjectVersion);
```

Comment:
We are assuming this exists (which is okay), just do projectSubscriptionUpdate.DataSourceVersions[ProjectDataSources.ConfiguredProjectVersion] so that we throw a KeyNotFoundException at the source if that assumption breaks instead of a null reference later when we access it.

Reply:
Done. Changed in this [commit](https://github.com/dotnet/project-system/pull/7199/commits/806af5662ebbfa17543d70b3e7edfafd9cd4d781).